app/models/card.rb

```diff
@@ -2,43 +2,68 @@ class Card < ApplicationRecord
   def self.search(filters)
     cards = Card.all
 
-    filters.each do |filter|
-      key, value = filter.first
-      cards = cards.where("#{key} ILIKE ?", "%#{value}%")
+    # This searches for cards by name iteratively
+    # for each name in the query string
+    filters[:name]&.each do |name|
+      cards = cards.where(
+        "name ILIKE ?",
+        "%#{name}%"
+      )
     end
 
-    cards
+    # This uses the ILIKE operator to search by
+    # name, description, etc. in a case-insensitive manner
+    %i[description rarity artist_name set
```

Comment:
Very nice. My only question is why not just include name in this as well? Then you can take out lines 7 to 12.